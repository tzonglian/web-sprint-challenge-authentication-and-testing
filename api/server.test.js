const request = require("supertest");
const server = require("./server.js");

describe("server.js module", () => {
  it("is the testing environment", () => {
    expect(process.env.DB_ENV).not.toBe("development");
    expect(process.env.DB_ENV).not.toBe("production");
    expect(process.env.DB_ENV).toBe("testing");
  });

  describe("[GET] /", () => {
    it("works", () => {
      return request(server)
        .get("/")
        .expect("Content-Type", /json/)
        .expect("Content-Length", "36")
        .expect({ api: "API is up - Dad joke time!" });
    });

    it("works also using jest syntax", async () => {
      const res = await request(server).get("/?foo=bar");
      expect(res.status).toBe(200);
      expect(res.body).toMatchObject({ api: "API is up - Dad joke time!" });
      expect(res.type).toMatch(/json/);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe("[GET] /api/jokes", () => {
    it("works and throws err bc no token", () => {
      return request(server)
        .get("/api/jokes")
        .expect("Content-Type", /json/)
        .expect({
          api: 'message: "Connection refused bc you do not have a token"',
        });
    });
  });
});
