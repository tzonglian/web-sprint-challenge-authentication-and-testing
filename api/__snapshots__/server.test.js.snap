// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server.js module [GET] / works also using jest syntax 1`] = `
Object {
  "api": "API is up - Dad joke time!",
}
`;
